app/helpers/enterprises_helper.rb

```diff
@@ -14,7 +14,7 @@ def current_customer
   def available_shipping_methods
     return [] if current_distributor.blank?
 
-    shipping_methods = current_distributor.shipping_methods
+    shipping_methods = current_distributor.shipping_methods.display_on_checkout.to_a
```

Comment:
Why `to_a`?

Reply:
otherwise it's a relation and it was not working with it. 
I cant remember exactly the error but I think it's related to some magic inside TagRuleApplicator (called just after this).